--- a/src/dfu.c
+++ b/src/dfu.c
@@ -95,7 +95,6 @@
		return -1;
	}

-	irecv_init();
	if (irecv_open_with_ecid(&dfu, client->ecid) != IRECV_E_SUCCESS) {
		return -1;
	}
@@ -119,7 +118,6 @@

	irecv_device_t device = NULL;

-	irecv_init();
	if (irecv_open_with_ecid(&dfu, client->ecid) != IRECV_E_SUCCESS) {
		return NULL;
	}
--- a/src/recovery.c
+++ b/src/recovery.c
@@ -108,7 +108,6 @@
		return -1;
	}

-	irecv_init();
	recovery_error=irecv_open_with_ecid(&recovery, client->ecid);

	if (recovery_error != IRECV_E_SUCCESS) {
