--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -112,4 +112,10 @@
     cflags_cc += common_flags
     cflags_objcc += common_flags
+    if (!is_clang) {
+      cflags_cc += [
+        "-include",
+        "cstring",
+      ]
+    }
 
     # Stack protection.
@@ -740,11 +746,7 @@
     "-Wno-unused-parameter",  # Unused function parameters.
   ]
 
-  if (!is_qnx) {
-    default_warning_flags += [
-      "-Werror",  # Warnings as errors.
-    ]
-  }
+  # Gentoo: do not treat warnings as errors.
 
   if (is_wasm) {
     default_warning_flags += [
