--- a/src/caelestia/subcommands/shell.py
+++ b/src/caelestia/subcommands/shell.py
@@ -1,5 +1,7 @@
 import subprocess
 from argparse import Namespace
+import os
+import sys

 from caelestia.utils.paths import c_cache_dir

@@ -26,7 +28,11 @@
             self.message(*self.args.message)
         else:
             # Start the shell
-            args = ["qs", "-c", "caelestia", "-n"]
+            if os.path.exists("/usr/share/quickshell/caelestia/shell.qml"):
+                args = ["qs", "-p", "/usr/share/quickshell/caelestia", "-n"]
+            else:
+                sys.exit("gui-apps/caelestia-shell is not installed. Please emerge gui-apps/caelestia-shell.")
+
             if self.args.log_rules:
                 args.extend(["--log-rules", self.args.log_rules])
             if self.args.daemon:
@@ -42,7 +48,11 @@
                             print(line, end="")

     def shell(self, *args: str) -> str:
-        return subprocess.check_output(["qs", "-c", "caelestia", *args], text=True)
+        if os.path.exists("/usr/share/quickshell/caelestia/shell.qml"):
+            qs_args = ["qs", "-p", "/usr/share/quickshell/caelestia"]
+        else:
+            sys.exit("gui-apps/caelestia-shell is not installed. Please emerge gui-apps/caelestia-shell.")
+        return subprocess.check_output([*qs_args, *args], text=True)

     def filter_log(self, line: str) -> bool:
         return f"Cannot open: file://{c_cache_dir}/imagecache/" not in line
--- a/src/caelestia/subcommands/screenshot.py
+++ b/src/caelestia/subcommands/screenshot.py
@@ -1,6 +1,8 @@
 import subprocess
 from argparse import Namespace
 from datetime import datetime
+import os
+import sys

 from caelestia.utils import hypr
 from caelestia.utils.notify import notify
@@ -21,8 +23,12 @@

     def region(self) -> None:
         if self.args.region == "slurp":
+            if os.path.exists("/usr/share/quickshell/caelestia/shell.qml"):
+                qs_args = ["qs", "-p", "/usr/share/quickshell/caelestia"]
+            else:
+                sys.exit("gui-apps/caelestia-shell is not installed. Please emerge gui-apps/caelestia-shell.")
             subprocess.run(
-                ["qs", "-c", "caelestia", "ipc", "call", "picker", "openFreeze" if self.args.freeze else "open"]
+                [*qs_args, "ipc", "call", "picker", "openFreeze" if self.args.freeze else "open"]
             )
         else:
             sc_data = subprocess.check_output(["grim", "-l", "0", "-g", self.args.region.strip(), "-"])
